### PR TITLE
Add initial prometheus support to redoctober.

### PR DESCRIPTION
**Pull request by kisom**
This doesn't add any redoctober-specific metrics yet; it's primarily intended to be a healthcheck at this point.

### CODE REVIEW

**Comment by kisom** [comment]
Also tested with a large test run in an attempt to find any intermittent issues:

```
$ go test -count 100 -timeout 2h
PASS
ok      github.com/cloudflare/redoctober        1970.774s
```

**Comment by lziest** [comment]
LGTM

File: redoctober.go [hunk 1]
```diff
@@ -203,6 +204,21 @@ func (this *indexHandler) handle(w http.ResponseWriter, r *http.Request) {
...
+// be started, a log.Fatal call is made.
+func initPrometheus() {
+	srv := &http.Server{
+		Addr:    net.JoinHostPort("localhost", metricsPort),
+		Handler: prometheus.Handler(),
+	}
+
+	log.Printf("metrics.init start: addr=%s", srv.Addr)
+	go func() {
+		log.Fatal(srv.ListenAndServe())
```
Comment (jkroll-cf) [inline redoctober.go]: This might be simpler:
```
log.Fatal(http.ListenAndServe(*metricsAddr, prometheus.Handler()))
```

File: redoctober.go [hunk 2]
```diff
@@ -242,6 +259,7 @@ func init() {
 	flag.StringVar(&hcKey, "hckey", "", "Hipchat API Key")
 	flag.StringVar(&hcRoom, "hcroom", "", "Hipchat Room Id")
 	flag.StringVar(&keysPath, "keys", "", "Path(s) of TLS private key in PEM format, comma-separated, must me in the same order as the certs")
+	flag.StringVar(&metricsPort, "metrics", "8081", "Port to use for metrics.")
```
Comment (jkroll-cf) [inline redoctober.go]: Any reason not to make the host configurable, too? What about machines with multiple external interfaces or service IPs?

File: redoctober.go (hunk 1 above)
Comment (jkroll-cf) [inline redoctober.go]: Ah. So the code I copied that from, which does work, puts it in a goroutine, like so:
```
go serveMetrics()
func serveMetrics() {
  log.Fatal(http.ListenAndServe(*metricsAddr, prometheus.Handler()))
}
```
Not sure why that's necessary, but the fact that ListenAndServe doesn't return unless there's an error is probably it.

File: redoctober_test.go [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+		Password: createVaultInput.Password,
+		Owners:   []string{createUserInput1.Name, createUserInput2.Name},
+		Data:     []byte("V2h5IGlzIGEgcmF2ZW4gbGlrZSBhIHdyaXRpbmcgZGVzaz8K"),
+	}
+)
+
+func init() {
+	// Because the certificate being used is self-signed, InsecureSkipVerify must be enabled
+	// to avoid the POST requests from failing.
+	cfg := &tls.Config{
```
Comment (jkroll-cf) [inline redoctober_test.go]: Is it not possible to simply inline this declaration?
```
http.DefaultClient.Transport = &http.Transport{
  TLSClientConfig: &tls.Config{
    InsecureSkipVerify: true,
  },
}
```
That would be cleaner, I think.

File: redoctober_test.go [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+	"testing"
+	"time"
+
+	"github.com/cloudflare/redoctober/core"
+)
+
+const baseURL = "https://localhost:8080/"
+
+var (
+	// Inputs that are used in multiple test cases are put here to reduce verbosity.
```
Comment (jkroll-cf) [inline redoctober_test.go]: I suspect this test file could be made easier to read and understand by moving to a table-driven test model (https://github.com/golang/go/wiki/TableDrivenTests). You'd have a struct representing each test case, and a single loop using your current functions to execute a test over the struct.

File: redoctober_test.go (hunk 3 above)
Comment (jkroll-cf) [inline redoctober_test.go]: Ah. Forgot this is the old file. We can get to it later.

File: redoctober_test.go (hunk 4 above)
Comment (jkroll-cf) [inline redoctober_test.go]: Forgot this was the old test file. We can fix this another time.

**Pull request merged**